.github/workflows/test.yaml

```diff
@@ -0,0 +1,66 @@
+name: Test
+
+on:
+    pull_request:
+        types:
+            - opened
+            - synchronize
+            - unlabeled
+
+permissions:
+    id-token: write
+    contents: read
+    deployments: write
+    pull-requests: write
+
+concurrency:
+    group: ${{ github.head_ref }}
+    cancel-in-progress: true
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v3
+
+            - name: Install pnpm
+              uses: pnpm/action-setup@v2
+              with:
+                  version: 8.6.0
+
+            - name: Setup Node.js with pnpm cache
+              uses: actions/setup-node@v3
+              with:
+                  node-version: '16'
+                  cache: 'pnpm'
+
+            - name: Install dependencies
+              run: pnpm install --frozen-lockfile
+
+            - name: Build
+              env:
+                HUSKY: 0
+
+                # Frontend environment variables.
+                VITE_ENDPOINT: ${{ env.apiSubDomain }}
+                NODE_ENV: ${{ env.nodeEnv }}
+
+                # Backend environment variables.
+                HOSTED_ZONE_ID: ${{ secrets.HOSTED_ZONE_ID }}
+                CERTIFICATE_ARN: ${{ secrets.CERTIFICATE_ARN }}
+                MONGODB_URI_PROD: ${{ secrets.MONGODB_URI_PROD }}
+                PR_NUM: ${{ github.event.pull_request.number }}
+
+                # CDK environment variables.
+                API_SUB_DOMAIN: ${{ env.apiSubDomain }}
+
+                # Turborepo credentials.
+                TURBO_API: ${{ vars.TURBO_API }}
+                TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
+                TURBO_TEAM: "team_antalmanac"
+              run: pnpm build
+
+            - name: Test
+              run: echo "all tests passed!!" # pnpm test run
```

Comment:
Why does this not do the real thing? Can I change it to actually running the tests??

Reply:
Idk, you tell me lol 
![image](https://github.com/icssc/AntAlmanac/assets/66761366/da092bc8-5318-4d69-a8dd-530d635a20bf)

Reply:
If you run `pnpm test` locally on this branch, there is no such test as `download-ics.test.ts`, this only exists on the main branch. And if you did put in the test on this branch, it runs perfectly fine ??

Reply:
idk man, works on my machine :tm: without putting that test in. I wanna see for myself what happens when we put it through the CI/CD again.

Reply:
That's literally exactly what I'm saying happens lol. It works locally, but not in CI/CD.

Reply:
Mb I thought you meant that it failed locally when you said
> If you run pnpm test locally on this branch, there is no such test as download-ics.test.ts

as in it failed with an error message like "no such test as download..."
Anyways I think we should fix the root cause, which seems to be that the build gets cached from a different branch, before merging this in, otherwise we're gonna see this problem over and over on every new PR

Reply:
Ok hang on. Bruh, we're using Node 16 https://github.com/icssc/AntAlmanac/blob/ci-tests/.github/workflows/test.yaml#L36; the current Node 18 (LTS) is the minimum that polyfills fetch globally. Let me try using that instead

Reply:
Ok, we're in the clear. I love devops 🥲

Reply:
I figured out the problem.
![image](https://github.com/icssc/AntAlmanac/assets/48658337/cb232d92-4722-4d29-b932-6c274e7562c1)
d1423 is the current `main` HEAD, and `609292` is one of the commits on this branch, when you activated the CI last. When it checks out the branch, it automatically is merging into main.

Reply:
Yeah I guess that would solve that test specifically failing in Actions, but now we also know why it's running tests that aren't on the branch the PR is based on.

Reply:
Ok cool, you can feel free to clean up the Git history if you want (or push some other miscellaneous change). I can merge it once you're done with everything :+1:

Reply:
Sounds good, I just pushed an update to the README and after the checks pass feel free to approve and merge.